src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ConnectedDevicesSpec.groovy

```diff
@@ -1383,6 +1384,174 @@ srcDevices=#newSrcEnabled, dstDevices=#newDstEnabled"() {
+        def tg = topology.activeTraffGens[0]
+        def sw = tg.switchConnected
+        def initialProps = switchHelper.getCachedSwProps(sw.dpId)
+        switchHelper.updateSwitchProperties(sw, initialProps.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = true
+            it.switchArp = true
+        })
+
+        and: "Flow is created on a target switch with devices feature 'off'"
+        def dst = topology.activeSwitches.find { it.dpId != sw.dpId }
+        def flow = flowHelperV2.randomFlow(sw, dst)
+        flow.source.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flow.destination.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flowHelperV2.addFlow(flow)
+
+        when: "Turn LLDP and ARP detection off on switch"
+
+        switchHelper.updateSwitchProperties(sw, initialProps.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = false
+            it.switchArp = false
+        })
+
+        flow.maximumBandwidth = flow.maximumBandwidth +1
+        flowHelperV2.updateFlow(flow.flowId, flow)
+
+        then: "CHeck excess rules are not registered on device"
+        wait(WAIT_OFFSET) {
+            { verifySwitchRules(sw.dpId) }
+        }
+
+
+        cleanup: "Remove created flow and registered devices, revert switch props"
+        flow && flowHelperV2.deleteFlow(flow.flowId)
+        database.removeConnectedDevices(sw.dpId)
+        initialProps && switchHelper.updateSwitchProperties(sw, initialProps)
+    }
+
+    @Tidy
+    def "System starts detect connected after device properties turned on"() {
+        given: "A switch with devices feature turned off"
+        def tgService = traffExamProvider.get()
+        assumeTrue(topology.activeTraffGens.size() > 0, "Require at least 1 switch with connected traffgen")
+        def sw = topology.activeTraffGens[0].switchConnected
+        def initialPropsSource = enableMultiTableIfNeeded(true, sw.dpId)
+        switchHelper.updateSwitchProperties(sw, initialPropsSource.jacksonCopy().tap {
+            it.multiTable = true
+            it.switchLldp = false
+            it.switchArp = false
+        })
+
+        and: "Flow is created on a target switch with devices feature 'off'"
+        def dst = topology.activeSwitches.find { it.dpId != sw.dpId }
+        def initialPropsDst = enableMultiTableIfNeeded(true, dst.dpId)
+        def flow = flowHelperV2.randomFlow(sw, dst)
+
+        def outerVlan = 100
+        flow.source.vlanId = outerVlan
+        flow.source.innerVlanId = 200
+
+        flow.source.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flow.destination.detectConnectedDevices = new DetectConnectedDevicesV2(false, false)
+        flowHelperV2.addFlow(flow)
+        northboundV2.getConnectedDevices(flow.source.switchId).ports.empty
+        northboundV2.getConnectedDevices(flow.destination.switchId).ports.empty
```

Comment:
You are checking that there is no connected devices, but you didn't send any connected device packets. Do you want just check that there were no devices before test stated?

Reply:
right - the reason to add these assertions here is to check the packets are not registeres before the option is applied/packets started